PopulateTools/gobierto
# New user Registration, Confirmation and session management workflow

Connects to #126.

### What does this PR do?

This PR implements the latest specs regarding User registration and session management. In brief, it adds a new Registration and Confirmation workflow, which can be accessed right from the login screen. This updated UI is also providing a way for requesting a password change.

It is also localising and translating any related template by keeping conventions established in the `GobiertoAdmin` module.

#### What is missing?

Once again, I'd need some help with CA translations. I'm keeping them mirrored from ES versions to leave a placeholder to work with 🙏 

### How should this be manually tested?

Just get into the new login screen at http://madrid.gobierto.dev/user/login, and check that:

- [x] Any User can recover her password.
- [x] Any Visitor can register in the new workflow: 1. Ask for email, 2. Ask for the remaining data.
- [x] Any User can start a session with valid credentials.
- [x] Every interaction is localised and translated into the current locale (please note the CA translations are not done yet).
- [x] The UI looks and behaves like the prototype.

Here're a couple of shots on those:

![screen shot 2016-12-14 at 14 25 42](https://cloud.githubusercontent.com/assets/126392/21185682/f86bd6fe-c211-11e6-9308-6c82e5d74957.jpg)

![screen shot 2016-12-14 at 14 27 36](https://cloud.githubusercontent.com/assets/126392/21185687/fcee2e0c-c211-11e6-9eef-c60088833ef3.jpg)
